src/algorithms/interface.ts

```diff
@@ -1,5 +1,3 @@
-interface DrawAlgorithm {
-
+export interface DrawAlgorithm {
```

Comment:
Yeah, in TS we have to export objects/interfaces. I learnt this recently, good catch.